src/pages/library/common.js

```diff
@@ -503,12 +503,11 @@ export const SearchAndFilterComponent = ({
       div({ style: { width: '19rem', flex: 'none' } }, [
         h(Sidebar, {
           onSectionFilter: section => setSelectedSections(_.xor([section])),
-          onTagFilter: ({ section, lowerTag, label }) => {
+          onTagFilter: ({ section, lowerTag }) => {
             Ajax().Metrics.captureEvent(`${Events.catalogFilter}:sidebar`, { tag: lowerTag })
-            setSelectedTags(oldSelectedTags => {
-              const newTags = _.xorBy('lowerTag', [{ lowerTag, label, section }], oldSelectedTags[section])
-              return newTags.length > 0 ? _.set(section, newTags, oldSelectedTags) : _.omit(section, oldSelectedTags)
-            })
+            const newTags = _.xorBy('lowerTag', [{ lowerTag }], selectedTags[section])
+            const newSelectedTags = newTags.length > 0 ? _.set(section, newTags, selectedTags) : _.omit(section, selectedTags)
+            navigateToFilterAndSelection(searchFilter, newSelectedTags)
```

Comment:
Its a little weird to need to pass the searchFilter in here (and the existing tags on above.) However, on looking at it, I think this is still preferable to two methods with one arg that call a common helper for the navigation event.

Reply:
Is there a way to provide named parameters with defaults in javascript? I came across this post https://simonsmith.io/destructuring-objects-as-function-parameters-in-es6#allowing-the-configuration-object-to-be-optional but couldn't get it to work.

Reply:
yes